src/ignore.c

```diff
@@ -336,14 +336,13 @@ int git_ignore__for_path(
 	}
 
 	if ((error = git_repository_item_path(&infopath,
-			repo, GIT_REPOSITORY_ITEM_INFO)) < 0)
+		repo, GIT_REPOSITORY_ITEM_INFO)) < 0 && error != GIT_ENOTFOUND)
 		goto cleanup;
 
 	/* load .git/info/exclude */
-	error = push_ignore_file(
+	if (error != GIT_ENOTFOUND && (error = push_ignore_file(
 		ignores, &ignores->ign_global,
-		infopath.ptr, GIT_IGNORE_FILE_INREPO);
-	if (error < 0)
+		infopath.ptr, GIT_IGNORE_FILE_INREPO)) < 0)
```

Comment:
I think this'd be easier to understand if you refactored it like you did refactor the attr code:

```
if ((error = git_repository_item_path(&infopath, ...)) < 0) ||
    (error = push_ignore_file(ignores, ...)) < 0) {
        if (error != GIT_ENOTFOUND)
            goto cleanup;
        error = 0;
}